src/ImageProcessor/Imaging/Filters/Photo/MatrixFilterBase.cs

```diff
@@ -29,41 +30,36 @@ public abstract class MatrixFilterBase : IMatrixFilter
         /// <param name="source">The current image to process</param>
         /// <param name="destination">The new image to return</param>
         /// <returns>
-        /// The processed <see cref="System.Drawing.Bitmap"/>.
+        /// The processed <see cref="System.Drawing.Bitmap" />.
         /// </returns>
         public abstract Bitmap TransformImage(Image source, Image destination);
 
         /// <summary>
-        /// Determines whether the specified <see cref="IMatrixFilter" />, is equal to this instance.
+        /// Determines whether the specified <see cref="System.Object" />, is equal to this instance.
         /// </summary>
-        /// <param name="obj">The <see cref="IMatrixFilter" /> to compare with this instance.</param>
+        /// <param name="obj">The <see cref="System.Object" /> to compare with this instance.</param>
         /// <returns>
-        ///   <c>true</c> if the specified <see cref="IMatrixFilter" /> is equal to this instance; otherwise, <c>false</c>.
+        ///   <c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.
         /// </returns>
-        public override bool Equals(object obj)
-        {
-            if (!(obj is IMatrixFilter filter))
-            {
-                return false;
-            }
+        public override bool Equals(object obj) => obj is IMatrixFilter matrixFilter && this.Equals(matrixFilter);
 
-            return this.GetType().Name == filter.GetType().Name
-                   && this.Matrix.Equals(filter.Matrix);
-        }
+        /// <summary>
+        /// Indicates whether the current object is equal to another object of the same type.
+        /// </summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns>
+        /// true if the current object is equal to the <paramref name="other" /> parameter; otherwise, false.
+        /// </returns>
+        public bool Equals(IMatrixFilter other) => other != null
+            && this.GetType() == other.GetType()
+            && this.Matrix == other.Matrix;
```

Comment:
Comparing `ColorMatrix` is always done using object equality. To make this work, we would need to check all 25 matrix values.